src/ape/contracts/base.py

```diff
@@ -463,6 +463,32 @@ def __len__(self):
         logs = self.provider.get_contract_logs(self.log_filter)
         return sum(1 for _ in logs)
 
+    def __call__(self, *args: Any, **kwargs: Any) -> "MockContractLog":
+        # Create a dictionary from the positional arguments
+        event_args = dict(zip((input.name for input in self.abi.inputs), args))
+
+        # Check for overwrites by looking for the intersection of the two key sets
+        overlapping_keys = set(event_args.keys()) & set(kwargs.keys())
+        if overlapping_keys:
+            raise ValueError(f"Overlapping keys found in arguments: {overlapping_keys}")
+
+        # Update event_args with keyword arguments
+        event_args.update(kwargs)  # type: ignore[arg-type]
+
+        # Check that event_args.keys() is a subset of the expected input names
+        expected_input_names = {input.name for input in self.abi.inputs}
+        if not set(event_args.keys()).issubset(expected_input_names):
+            raise ValueError(
+                f"Invalid argument keys found, expected a subset of {expected_input_names}"
```

Comment:
`setA - setB` gives you items in `setA` that are not in `setB`
```suggestion
        if unknown_input_names := set(event_args.keys()) - {input.name for input in self.abi.inputs}:
            raise ValueError(
                f"Invalid argument keys found, expected a subset of {unknown_input_names}"
```
p.s. I enjoy walrus operators way too much